.dockerignore

```diff
@@ -0,0 +1 @@
+vendor/
```

Comment:
Can we get rid of this file?

Reply:
Why? This ensures that dep will always download the required dependencies according to the `Gopkg.lock` file in the repo. It also makes the build context smaller.

Reply:
Can you explain a little bit why do we need this? 
Seems like the way it works changes. It used to clone the repo and build from there. But here  you are building from current repo, right? But I still don't understand what this is for. Because dep always download from Gopkg.lock. "makes the build contextw smaller" do you mean the it relates to the image?

Reply:
Yeah, one of the biggest changes here is instead of cloning the repo, docker uses the working copy to build the image. This is aligned with the current best practices of docker image building.

Files listed in docker ignore won't be copied into the image by `ADD` and `COPY` commands. Ignored files are also not passed as build context, which is the list of files and their content. When building an image, the commands in the dockerfile use the so called build cache. When a command's result will be the same as another command from a previous build (eg. copy file A into the image. if fileA has the same content as in a previous build, docker will use that layer from the already built image, instead of copying the file again). This is an oversimplified explanation of what build context is and how it relates to build cache. To put it simple: it makes the image build slightly faster, but does not have an effect on the image size.

The reason why vendor is ignored is to make sure as much building happens within the docker context as possible. Basically, everything that's not source code copying, should happen within docker.

Another reason is the aforementioned build cache. This way if `Gopkg.lock` didn't change since the last build, docker will simply use an earlier layer from the cache, gaining some speed and storage, but again, it does not affect the final image size.

The thing is: it's hard to explain why these practices make sense and why they make things easier without an in-depth knowledge of Docker, so I would suggest reading the best practices in the Docker documentation, because I'm pretty sure it explains these things much-much better, than I do.

Reply:
Got it. That makes a lot of sense to me. Thanks!